Basic stats

RavenX8 opened the pull request:
- Added level up and exp gain
- Added movement speed calculations
- Added attack speed calculations
- Updated item lua
- Fixed crash when picking up zuly
- Fixed windows compile issues
- Updated build system to be more robust

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/dev-osrose/osirose-new/210)
<!-- Reviewable:end -->

rminderhoud committed:
Add some more packets

rminderhoud committed:
Move togglemove enums to packets

RavenX8 committed:
added damage log support
Added exp calcs for killing mobs

RavenX8 committed:
make sure our vars are reset

RavenX8 committed:
Merge remote-tracking branch 'origin/trunk' into basic-stats

RavenX8 committed:
Exp calc done

RavenX8 committed:
Fixed issue where the client would see MISS after damage numbers

RavenX8 committed:
NPCs now have their level data assigned

RavenX8 committed:
Merge remote-tracking branch 'origin/trunk' into basic-stats

RavenX8 committed:
Flush the logs before dropping the sinks

RavenX8 committed:
Merge remote-tracking branch 'origin/ralph/packets' into basic-stats

RavenX8 committed:
renamed xml file

RavenX8 committed:
made some changes to the packets

RavenX8 committed:
Added exp gain and level up support

RavenX8 committed:
Updated spdlog

RavenX8 committed:
Updated packet generator module to match other modules

RavenX8 committed:
Merge remote-tracking branch 'origin/trunk' into basic-stats

RavenX8 committed:
Cleaned up 3rdparty cmakelists.txt

RavenX8 committed:
Cleaned up some cmake vars that weren't used

RavenX8 committed:
Updated version info

RavenX8 committed:
More cleanup

RavenX8 committed:
Fix issue where colored logging wasn't working

RavenX8 committed:
since slack reissued the webhook key, lets just remove it

RavenX8 committed:
Merge remote-tracking branch 'origin/trunk' into basic-stats

RavenX8 committed:
more updates to build system

RavenX8 committed:
Some work on #124

RavenX8 committed:
Fixed typo in CMakeLists.txt

RavenX8 committed:
Updated spdlog to get windows compile error fix

RavenX8 committed:
Syncing work for windows

RavenX8 committed:
Fixed typo for mysql lib when building it with cmake
Updated python find_package name to use the non-deprecated version

RavenX8 committed:
Fix comment typo

RavenX8 committed:
Added cmake module to build zlib for CURL

RavenX8 committed:
Added windows support for building mysql

RavenX8 committed:
More updates to building mysql on windows.

RavenX8 committed:
Added some functions and stoof

RavenX8 committed:
Merge remote-tracking branch 'origin/trunk' into basic-stats

RavenX8 committed:
Added 3 new functions to the item lua script

RavenX8 committed:
Merge branch 'basic-stats' of https://github.com/dev-osrose/osIROSE-new into basic-stats

RavenX8 committed:
Merge remote-tracking branch 'origin/trunk' into basic-stats

RavenX8 committed:
run speed is now calculated instead of being hardcoded

RavenX8 committed:
Merge remote-tracking branch 'origin/trunk' into basic-stats

RavenX8 committed:
Added attack speed calc function

RavenX8 committed:
Fixed typo in function call

RavenX8 committed:
Attempted to make it possible for the server to upload the dmp file to a url at crash time

RavenX8 committed:
Fixed a crash when trying to pick up zuly from the ground

RavenX8 committed:
Fixed issue where when unequipping a weapon, your attack speed would be much slower then intended

RavenX8 committed:
Added more info to the dump upload report

RavenX8 committed:
Enabled symbol uploads for appveyor

RavenX8 committed:
updated symbol files for appveyor

RavenX8 committed:
since I clearly can't make a appveyor change blind, I'm enabling the debug connection

RavenX8 committed:
Put url in quotes

RavenX8 committed:
Updated README

RavenX8 committed:
Merge remote-tracking branch 'origin/trunk' into basic-stats

RavenX8 committed:
Updated README [ci skip]

RavenX8 committed:
update min required version of cmake

RavenX8 committed:
Merge remote-tracking branch 'origin/trunk' into basic-stats

RavenX8 committed:
Fixed annoying issue where if we found a prev breakpad build, the find module wouldn't create the interface target correctly causing a build error when using ninja

L3nn0x commented on README.md:
```diff
@@ -72,7 +90,7 @@ If you opted to keep them in place, the server binaries are located in the bin f
     $ ./bin/MapServer&
 
-After running for the first time, the servers will create a config file in the current directory.
+After running for the first time, the servers will create a config file in the current directory that you need to edit.
```
We probably want to explain what each option does and how you can modify it. A help page in the github wiki?

L3nn0x commented on src/map/src/utils/calculation.cpp:
```diff
@@ -0,0 +1,107 @@
+      if(entitySystem.has_component<Component::Inventory>(entity) == true) {
+        auto& inventory = entitySystem.get_component<Component::Inventory>(entity);
+        const auto& boots = inventory.boots();
+        if(boots) {
```
replace those if with the getter in utils/getters.h

RavenX8 committed:
update appveyor settings

RavenX8 committed:
Fix VS path issue on appveyor

RavenX8 committed:
Removing some build system macro usage that isn't recommened

RavenX8 committed:
Fixed compile error on windows if crash reports are disabled

RavenX8 committed:
General improvements to building breakpad on windows.

RavenX8 committed:
Merge branch 'basic-stats' of github.com:dev-osrose/osIROSE-new into basic-stats

RavenX8 committed:
Made a send queue change to prevent stoof from breaking hopefully

RavenX8 committed:
Removed useless compile flag for unit tests

RavenX8 committed:
Fix compile error with printing the packets out

RavenX8 committed:
Fixed issue where map server would mark an account as offline when going to character select

RavenX8 committed:
Changed timeout value from 1 minute to 5 minutes

The server sending packets on the socket no longer updates the last recvd time.

RavenX8 committed:
Attack speed change

RavenX8 committed:
Fixed issue where the SQLPPGenerator wasn't using the found python exec and was instead relying on it being in the path

RavenX8 committed:
Update README.md

RavenX8 committed:
Update README.md

RavenX8 committed:
Merge branch 'basic-stats' of github.com:dev-osrose/osIROSE-new into basic-stats

RavenX8 committed:
Merge remote-tracking branch 'origin/trunk' into basic-stats

RavenX8 committed:
Fix github labeler yml issue

RavenX8 committed:
Added more paths to labeler

RavenX8 committed:
Removed unused CMakeLists.txt
Updated cache directories for CI providers
Moved build time 3rdparty install directory into the build directory by default (can be changed by setting CMAKE_THIRD_PARTY_DIR)
Updated FindMysql.cmake to look for MariaDB
Updated BreakpadSymbolGeneration.cmake and BuildGoogleBreakpad.cmake to make windows builds play nice
Added robocopy_flat.bat to copy files but not their directory structure
Disabled copying of libmysql.dll into the bin directory as it was causing issues when using MariaDB instead of MySQL

RavenX8 committed:
Reverted FindMySQL change

RavenX8 committed:
update lcov install script

RavenX8 committed:
Updated appveyor paths

RavenX8 committed:
Added link flags for gnu when building with coverage reports

RavenX8 committed:
updated appveyor build paths

RavenX8 committed:
removed webhook thingy which was annoying

RavenX8 committed:
enabled appveyor debugging

RavenX8 committed:
fixed appveyor issue

RavenX8 committed:
stop trying to upload files that arent there

RavenX8 committed:
add retry to curl command

RavenX8 committed:
more changes for appveyor to work

RavenX8 committed:
Upgraded the distro to bionic

RavenX8 committed:
Missed the global PATH env var

RavenX8 committed:
Merge remote-tracking branch 'origin/trunk' into basic-stats

RavenX8 committed:
Added new notify-discord script for linux.
Adjusted file permissions on other scripts

RavenX8 committed:
Updated travis builds with new discord script and new job names

RavenX8 committed:
Update .travis.yml

RavenX8 committed:
Fixed discord notification URL

RavenX8 committed:
Removed before_script command from travis

RavenX8 committed:
added some damage calculation functions

L3nn0x commented on unix_env_setup.sh:
```diff
@@ -2,7 +2,7 @@
 sudo apt-add-repository -y "ppa:ubuntu-toolchain-r/test"
 sudo apt-get update
-sudo apt-get install g++-8 unzip autoconf libtool libssl-dev python perl libncurses5-dev libreadline-dev
+sudo apt-get install g++-8 unzip autoconf libtool libssl-dev python libncurses5-dev libreadline-dev
```
g++-9 instead of g++-8 no?

RavenX8 replied:
no, I maintain this to match what we are using on travis. Travis uses g++-8

RavenX8 committed:
Updated calc stoof

L3nn0x commented on src/map/src/utils/calculation.cpp:
```diff
@@ -0,0 +1,147 @@
+
+  int get_maxhp(EntitySystem& entitySystem, RoseCommon::Entity entity) {
+    auto& life = entitySystem.get_component<Component::Life>(entity);
+    //TODO do calc here
```
which calc? It's life.mapHp

RavenX8 replied:
this is for calculating the max hp based on all stats and buffs. This isn't used yet

L3nn0x commented on src/map/src/utils/calculation.cpp:
```diff
@@ -0,0 +1,147 @@
+
+  int get_maxmp(EntitySystem& entitySystem, RoseCommon::Entity entity) {
+    auto& magic = entitySystem.get_component<Component::Magic>(entity);
+    //TODO do calc here
```
which calc? It's magic.maxMp

RavenX8 replied:
this is for calculating the max mp based on all stats and buffs. This isn't used yet.

L3nn0x commented on src/map/src/combat/combat.cpp:
```diff
@@ -215,56 +222,135 @@ void Combat::update(EntitySystem& entitySystem, Entity entity, uint32_t dt) {
+
+            //TODO:: adjust our hp and mp based on our level here
+            //TODO:: give the player attr and skill points
+            auto p = SrvLevelup::create(attack_log.attacker_, attackerLevel.level, attackerLevel.xp, 0, 0);
```
maybe have a dedicated function to take care of the whole level up thingy?

RavenX8 replied:
Will do this in next combat pass

L3nn0x commented on src/map/src/combat/combat.cpp:
```diff
@@ -215,56 +222,135 @@ void Combat::update(EntitySystem& entitySystem, Entity entity, uint32_t dt) {
-    
+
     if(((entitySystem.has_component<Component::Npc>(target.target) == false && entitySystem.has_component<Component::Mob>(target.target) == true) ||
       (false))) // TODO:: Check if this map has PVP turned on and the target player isn't on my team
```
maybe we want to have a function to check if the target is valid within a map

RavenX8 replied:
Will do this in next combat pass

L3nn0x commented on src/map/src/combat/combat.cpp:
```diff
@@ -418,6 +504,25 @@ void Combat::revive(EntitySystem& entitySystem, Entity entity, const RoseCommon:
   entitySystem.teleport_entity(entity, std::get<1>(dest), std::get<2>(dest), std::get<0>(dest));
 }
 
+int32_t Combat::get_exp_to_level(int level) {
```
prob should be a int64_t here no?

RavenX8 replied:
the level component is int32_t, only made that match that.

RavenX8 replied:
Anyway I have changed it in the latest

L3nn0x replied:
I had put in32_t here because the packets expect that 🤷‍♂

L3nn0x reviewed (commented):
looks ok

RavenX8 committed:
Updated get_exp type
